astronomer/providers/amazon/aws/operators/redshift_data.py

```diff
@@ -43,6 +43,30 @@ def execute(self, context: Context) -> None:  # type: ignore[override]
         if response.get("status") == "error":
             self.execute_complete(context, event=response)
         context["ti"].xcom_push(key="return_value", value=query_ids)
+
+        still_running = False
+        completed_query_ids = []
+        for qid in query_ids:
```

Comment:
I saw a similar change in PR https://github.com/astronomer/astronomer-providers/pull/1090 does it make sense to make this loop a function?

Reply:
Yep, I think that's something we can do. I just merge #1090  so that I can extract this logic in this PR

Reply:
@pankajastro just finished the refactor. could you please check? Thanks!

Reply:
looks good. thanks